fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/properties/RecordTypesProperty.java

```diff
@@ -152,4 +161,12 @@ public boolean shouldVisit(@Nonnull ExpressionRef<? extends RelationalExpression
     public static Set<String> evaluate(@Nonnull PlanContext context, @Nonnull RelationalExpression ref) {
         return ref.acceptPropertyVisitor(new RecordTypesProperty(context));
     }
+
+    @Nonnull
+    @Override
+    @SpotBugsSuppressWarnings("NP_PARAMETER_MUST_BE_NONNULL_BUT_MARKED_AS_NULLABLE")
+    public Set<String> evaluateAtQuantifier(@Nonnull final Quantifier quantifier, @Nullable final Set<String> rangesOverResult) {
+        // since we do visit expression references in this property we can insist on rangesOverResult not being null
```

Comment:
```suggestion
        // since we visit the expression reference under the quantifier, and don't return null ourselves, we can
        // insist that rangesOverResult is never null
```